### PR TITLE
Update helm.mdx

### PR DESCRIPTION
**Pull request by maksimnosal**
Added comments/requirements as per https://github.com/hashicorp/consul-dataplane/blob/7e075dc4310e609a31babe4cda5c24cfb0f6a416/cmd/consul-dataplane/main.go#L111

### PR Checklist

* [x] updated test coverage
* [x] external facing docs updated
* [x] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

File: website/content/docs/k8s/helm.mdx [hunk 1]
```diff
@@ -1104,12 +1104,13 @@ Use these links to navigate to a particular top-level stanza.
...
     should be the same, however, they may be different if you
     wish to use separate hosts for the HTTPS connections.
+    'tlsServerName' is required if TLS is enabled and 'hosts' is not a DNS name.
 
   - `httpsPort` ((#v-externalservers-httpsport)) (`integer: 8501`) - The HTTPS port of the Consul servers.
 
   - `grpcPort` ((#v-externalservers-grpcport)) (`integer: 8502`) - The GRPC port of the Consul servers.
 
-  - `tlsServerName` ((#v-externalservers-tlsservername)) (`string: null`) - The server name to use as the SNI host header when connecting with HTTPS.
+  - `tlsServerName` ((#v-externalservers-tlsservername)) (`string: null`) - The server name to use as the SNI host header when connecting with HTTPS. The hostname to expect in the server certificate's subject
```
Comment (david-yu) [inline website/content/docs/k8s/helm.mdx:1113]:
```suggestion
  - `tlsServerName` ((#v-externalservers-tlsservername)) (`string: null`) - The server name to use as the SNI host header when connecting with HTTPS. This also sets the hostname to expect in the server certificate's subject when communicating to external servers
```

Comment (david-yu) [reply website/content/docs/k8s/helm.mdx:1113]: Is this accurate?

Comment (maksimnosal) [reply website/content/docs/k8s/helm.mdx:1113]: Yes, it is correct. But the main part is under 'host' stanza: 'tlsServerName' is required if TLS is enabled and 'hosts' is not a DNS name.

Comment (david-yu) [inline website/content/docs/k8s/helm.mdx:1113]:
```suggestion
  - `tlsServerName` ((#v-externalservers-tlsservername)) (`string: null`) - The server name to use as the SNI host header when connecting with HTTPS. The hostname to expect in the server certificate's subject.
```

**Comment by david-yu** [comment]
@maksimnosal Could you sign the CLA?

**Comment by david-yu** [comment]
Closing as docs are updated here instead: https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/values.yaml

**Pull request closed without merging**